interface/main/messages/messages.php

```diff
@@ -23,614 +24,686 @@
+                    <a class='more' href=<?php echo $lnkvar1; ?>> <?php echo $linkvar2; ?> </a>
+                  </td>
+                </tr>
+              </table>
+                <?php
+              //show the activity links
+                if (empty($task) || $task=="add" || $task=="delete") { ?>
+                    <?php if ($active == "all") { ?>
+                  <span><?php echo xlt('Show All'); ?></span>
+                <?php } else { ?>
+                <a href="messages.php" class="link" onclick="top.restoreSession()"><span><?php echo xlt('Show All'); ?></span></a>
+                <?php } ?>
+                  |
+                    <?php if ($active == '1') { ?>
+                <span><?php echo xlt('Show Active'); ?></span>
+                <?php } else { ?>
+                <a href="messages.php?form_active=1" class="link" onclick="top.restoreSession()"><span><?php echo xlt('Show Active'); ?></span></a>
+                <?php } ?>
+                  |
+                    <?php if ($active == '0') { ?>
+                <span><?php echo xlt('Show Inactive'); ?></span>
+                <?php } else { ?>
+                <a href="messages.php?form_inactive=1" class="link" onclick="top.restoreSession()"><span><?php echo xlt('Show Inactive'); ?></span></a>
+                <?php } ?>
+                <?php }  ?>
+            </div>
+            <?php
+            switch($task) {
+                case "add" :
+                {
+                    // Add a new message for a specific patient; the message is documented in Patient Notes.
+                    // Add a new message; it's treated as a new note in Patient Notes.
+                    $note = $_POST['note'];
+                    $noteid = $_POST['noteid'];
+                    $form_note_type = $_POST['form_note_type'];
+                    $form_message_status = $_POST['form_message_status'];
+                    $reply_to = $_POST['reply_to'];
+                    $assigned_to_list = explode(';', $_POST['assigned_to']);
+                    foreach($assigned_to_list as $assigned_to){
+                        if ($noteid && $assigned_to != '-patient-') {
+                            updatePnote($noteid, $note, $form_note_type, $assigned_to, $form_message_status);
+                            $noteid = '';
+                        }
+                        else {
+                            if($noteid && $assigned_to == '-patient-'){
+                                // When $assigned_to == '-patient-' we don't update the current note, but
+                                // instead create a new one with the current note's body prepended and
+                                // attributed to the patient.  This seems to be all for the patient portal.
+                                $row = getPnoteById($noteid);
+                                if (! $row) die("getPnoteById() did not find id '".text($noteid)."'");//?translate
```

Comment:
since this is really a developer debugging level error, no need to translate this.